lua/neo-tree/sources/common/components.lua

```diff
@@ -147,4 +148,71 @@ M.name = function(config, node, state)
   }
 end
 
+local indent_cache = {}
+
+local indent_without_markers = function(config, node, state)
+  local level = node.level
+  local indent_size = state.indent_size or 2
+
+  if level == 0 or indent_size == 0 then
+    return {}
+  end
+
+  if not indent_cache[level] then
```

Comment:
I don't think this cache is worth the complication. What happens if the user changes the configuration? Then you need to check for that and clear the cache. Also, a user can have different indent settings per source. It would be better to just
```lua
return string.rep(" ", indent_size * level)
```